pkg/cstor/volume/v1alpha1/cstorvolume.go

```diff
@@ -209,3 +278,100 @@ func (c Conditions) UpdateCondition(cond apis.CStorVolumeCondition) []apis.CStor
 	}
 	return c
 }
+
+// BuildConfigData builds data based on the CVReplicationDetails
+func (csr *CVReplicationDetails) BuildConfigData() map[string]string {
+	data := map[string]string{}
+	// Since we know what to update in istgt.conf file so constructing
+	// key and value pairs
+	// key represents what kind of configurations
+	// value represents corresponding value for that key
+	// TODO: Improve below code by exploring different options
+	key := fmt.Sprintf("  ReplicationFactor")
+	value := fmt.Sprintf("  ReplicationFactor %d", csr.ReplicationFactor)
+	data[key] = value
+	key = fmt.Sprintf("  ConsistencyFactor")
+	value = fmt.Sprintf("  ConsistencyFactor %d", csr.ConsistencyFactor)
+	data[key] = value
+	key = fmt.Sprintf("  Replica %s", csr.ReplicaID)
+	value = fmt.Sprintf("  Replica %s %d", csr.ReplicaID, csr.ReplicaGUID)
+	data[key] = value
+	return data
+}
+
+// UpdateConfig updates target configuration file by building data
+func (csr *CVReplicationDetails) UpdateConfig() error {
+	configData := csr.BuildConfigData()
+	fileOperator := util.RealFileOperator{}
+	ConfFileMutex.Lock()
+	err := fileOperator.UpdateOrAppendMultipleLines(IstgtConfPath, configData, 0644)
+	ConfFileMutex.Unlock()
+	return err
+}
+
+// Validate verifies whether CStorReplication data read on wire is valid or not
+func (csr *CVReplicationDetails) Validate() error {
+	if csr.VolumeName == "" {
+		return errors.Errorf("volume name can not be empty")
+	}
+	if csr.ReplicaID == "" {
+		return errors.Errorf("replicaKey can not be empty to perform "+
+			"volume %s update", csr.VolumeName)
+	}
+	if csr.ReplicaGUID == 0 {
+		return errors.Errorf("replicaKey can not be empty to perform "+
+			"volume %s update", csr.VolumeName)
+	}
+	if csr.ReplicationFactor == 0 {
+		return errors.Errorf("replication factor can't be %d",
+			csr.ReplicationFactor)
+	}
+	if csr.ConsistencyFactor == 0 {
+		return errors.Errorf("consistencyFactor factor can't be %d",
+			csr.ReplicationFactor)
+	}
+	return nil
+}
+
+// UpdateCVWithReplicationDetails updates the cstorvolume with known replicas
+// and updated replication details
+func (csc *CStorVolumeConfig) UpdateCVWithReplicationDetails() error {
+	err := csc.Validate()
+	if err != nil {
+		return errors.Wrapf(err, "validate errors")
+	}
+	cv, err := csc.Get(csc.VolumeName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get cstorvolume")
+	}
+	if len(cv.Status.ReplicaDetails.KnownReplicas) >= cv.Spec.DesiredReplicationFactor {
```

Comment:
replica replacement case will get error here
Also, duplicate updates also will get error here, like, if update to CR succeeded but not returned success to istgt.